implement S4U (S4U2Proxy & S4U2Self)

dmansfield opened the pull request:
This pull request implements S4U2Proxy (constrained delegation) and S4U2Self (protocol transition) in the server side functions.  Behavior is optional.

No client side methods are touched at all.

There is one other feature that _does_ touch client code that I will submit a separate pull request for.  (I previously had mentioned two features but I have rolled both of the server side modifications into this pull request).

Commit:
implement S4U2Proxy

Commit:
implement S4U2Self

christkv commented:
Hi currently https://github.com/christkv/kerberos/pull/24 is priority number 1. Once I've merged I'll be happy to consider this pull request my only requirement is that is pass our kerberos tests for the driver. if it does I'll take it :+1:

Commit:
adjustments for nan 2.0

dmansfield commented:
Hey! I've rebased against master and updated to Nan 2.0 in the new code. I've tested functionality for server side non-S4U, S4UProxy and S4U2Self.  Let me know if you need anything else.

futurechan commented:
:+1:
